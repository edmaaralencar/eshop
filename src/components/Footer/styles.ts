import styled from 'styled-components'

export const Wrapper = styled.footer`
  margin-top: 3.2rem;
  padding: 2.4rem 0;

  text-align: center;
  color: #666666;
  font-size: 1.6rem;

  border-top: 3px solid ${({ theme }) => theme.colors.gray};
`
