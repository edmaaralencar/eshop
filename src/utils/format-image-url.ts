export function formatImageUrl(image: string) {
  return `/products/${image}`
}
