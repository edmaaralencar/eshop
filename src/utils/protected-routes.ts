import { GetServerSidePropsContext } from 'next'
// import { unstable_getServerSession } from 'next-auth'
import { getSession } from 'next-auth/react'
// import { options } from 'pages/api/auth/[...nextauth].api'

async function protectedRoutes(context: GetServerSidePropsContext) {
  // const session = await unstable_getServerSession(
  //   context.req,
  //   context.res,
  //   options
  // )

  // if (!session) {
  //   return {
  //     redirect: {
  //       destination: `/sign-in?callbackUrl=${context.resolvedUrl}`,
  //       permanent: false
  //     }
  //   }
  // }

  const session = await getSession(context)

  if (!session) {
    context.res.setHeader(
      'Location',
      `/sign-in?callbackUrl=${context.resolvedUrl}`
    )
    context.res.statusCode = 302
  }

  return session
}

export default protectedRoutes
