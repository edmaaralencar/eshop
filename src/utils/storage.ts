const APP_KEY = '@eshop'

export function getStorageItem(key: string) {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(`${APP_KEY}.${key}`)
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return JSON.parse(data!)
}

export function setStorageItem(key: string, value: any[]) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(`${APP_KEY}.${key}`, data)
}

export function removeStorageItem(key: string) {
  if (typeof window === 'undefined') return

  return window.localStorage.removeItem(`${APP_KEY}.${key}`)
}
