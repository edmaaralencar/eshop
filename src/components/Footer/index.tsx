import * as S from './styles'

function Footer() {
  return (
    <S.Wrapper>
      <span>Edmar Alencar 2022 Â© Todos os direitos reservados</span>
    </S.Wrapper>
  )
}

export default Footer
