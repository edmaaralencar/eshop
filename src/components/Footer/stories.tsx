import { StoryObj, Meta } from '@storybook/react'
import Footer from '.'

export default {
  title: 'Footer',
  component: Footer
} as Meta

export const Default: StoryObj = {}
