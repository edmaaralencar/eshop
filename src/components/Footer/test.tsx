import { render, screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test-helper'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Footer />)

    expect(
      screen.getByText('Edmar Alencar 2022 Â© Todos os direitos reservados')
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
