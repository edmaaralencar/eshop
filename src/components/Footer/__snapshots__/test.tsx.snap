// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the heading 1`] = `
.c0 {
  margin-top: 3.2rem;
  padding: 2.4rem 0;
  text-align: center;
  color: #666666;
  font-size: 1.6rem;
  border-top: 3px solid #f5f8fa;
}

<footer
  class="c0"
>
  <span>
    Edmar Alencar 2022 Â© Todos os direitos reservados
  </span>
</footer>
`;
